### ISSUE TITLE
Since Nuxt 2.9.0, Nuxt is really slow. (Works perfect on 2.8.1)

### ISSUE THREAD

**Issue by t1gu1** [nuxt/nuxt#6508]
### Version

[v2.9.0](https://github.com/nuxt.js/releases/tag/v2.9.0)

### Reproduction link

[https://github.com/t1gu1/nuxt-2.9.0-and-more-is-slower](https://github.com/t1gu1/nuxt-2.9.0-and-more-is-slower)

### Steps to reproduce

- Install the project and read readme.
- Switch between Menu 1 and Menu 2 few time.

We can really see the issue when we change from a page to another (Click on a menu link few time)

### What is expected ?

Be as fast as 2.8.1

### What is actually happening?

Since the 2.9.0 the performance drop is huge. (3s+ to load a page)
As is said, we can really see the issue when we change from a page to another (Click on a menu link few time)

### Additional comments?

On the 2.8.1, it seems take less than an half second. (Maybe less it's really fast)
I just test the 2.10.0 and the issue is still present. (We build and start a new project to test)

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9857">#c9857</a>)</em></sub></div>

**Comment by danielroe** [comment]
See #6279, with potential workaround.

**Comment by t1gu1** [comment]
Hi @danielroe,

I don't really see any workaround in the issue #6279 and the status is defined close without solution.
The only workaround i've found myself is to downgrade to the 2.8.1. :(

Thanks for the answer.

**Comment by danielroe** [comment]
@t1gu1 Worth a try - https://github.com/nuxt/nuxt.js/issues/6279#issuecomment-528763416: `export const strict = false` in `store/index.js`

**Comment by t1gu1** [comment]
Except the index.js with pretty nothing inside that I don't use, I don't really have a store in the reproductible project that I share.

**Comment by t1gu1** [comment]
@danielroe, I just remove store (index.js) and the speed of the application is effectively more constant and faster. It takes now 1.8s to load a page and not 3s. The speed seems decrease with time.

BUT with the version 2.8.1 it takes like 200ms(I dont really know) to load a page. It's really faster in 2.8.1.

The workaround is effective but really not as downgrade the version to 2.8.1.

**Comment by aldarund** [comment]
@t1gu1 does it happen only in dev mode or in production build too?

**Comment by clarkdo** [comment]
@t1gu1 Thank you so much for the report, I’ll look into it this weekend

**Comment by clarkdo** [comment]
Hi @t1gu1 , there is some founding from @aldarund , could you try to turn off vue-meta by setting features.meta: false in nuxt.config?

**Comment by aldarund** [comment]
@t1gu1 to workaround your issue you need to use nuxt-i18n better.  disable seo in nuxt config and see this -> https://github.com/nuxt-community/nuxt-i18n/blob/master/docs/seo.md#improving-performance

**Comment by t1gu1** [comment]
@aldarund, from my experience about that problem, this is just a tiny mini bit better in prod mode.
I'll repeat myself but it still really not comparable to the 2.8.1.

Also thanks @clarkdo & @aldarund for the vue-meta / seo reference.
**I'll test it** and **make a feedback to you in my next comment**.

I'll just let you keep in mind than that happend only **after 2.8.1** and it **was fine before**. (Testable in the linked project if you switch between version in the package.json)

So basically I think that your points are nice performance optimisation that i'll try to keep in mind for my future & current project (Thanks for that) **but** it's not **the** problem.

What's change in the 2.9.0 that could make all that slower and now require a lot performance optimisation from Nuxt users? What could you guess?

Thanks for all again!! :)

**Comment by danielroe** [comment]
@t1gu1 Here is the changelog: https://github.com/nuxt/nuxt.js/releases/tag/v2.9.0

Note the vue-meta upgrade, for example.

**Comment by AndrewBogdanovTSS** [comment]
@t1gu1 which version of vue-meta are you using? Do you also use `nuxt-jsonld` or `nuxt-i18n` ? If yes try to force install `vue-meta@2.3.0-beta.0` and let me know how it affected the performance.
Check out https://github.com/nuxt/vue-meta/issues/443 to get more context of an issue

**Comment by t1gu1** [comment]
**Best workaround** from @aldarund
:heavy_check_mark: :heavy_check_mark: :heavy_check_mark: :heavy_check_mark: :heavy_check_mark: :heavy_check_mark: 
@aldarund This is it!!! :heart:  #bigThanks
Your link about that: https://github.com/nuxt-community/nuxt-i18n/blob/master/docs/seo.md#improving-performance
:heavy_check_mark: :heavy_check_mark: :heavy_check_mark: :heavy_check_mark: :heavy_check_mark: :heavy_check_mark: 

**Since I turn seo off and add it as it's described(The optimized way), the speed is as good as 2.8.1. (And I keep the vuex store without the changes we talks)**

_p.s. Desactivate store make a little difference but it was really nothing compare to turn seo off and use the workaround._

-------------------------

@AndrewBogdanovTSS, I use `nuxt-i18n`.
`vue-meta` is a default nuxt package so i'm not sure how to make the force install of it.
If it's not a default package installed in Nuxt, I do not have that package in the test project that we talk. If it is. feel free to use my reproduction link to test your point. (Sry, I just dont really know how to force a package of a package)

-------------------------

To resume, the performance drop since the 2.9.0 is clearly about meta/nuxt-i18n meta/seo.
@AndrewBogdanovTSS, it really looks something like the issue you share (nuxt/vue-meta#443), because I really feel it when the routing is trigger and page change.

Thanks again @everyone!

**Comment by pimlie** [comment]
Please note that `vue-meta@2.3.0` has been released, no need to install the beta anymore. You can run `yarn upgrade` or `yarn add vue-meta@2.3.0` to install the newer version

@t1gu1 Last friday I spent quite some test trying to get to the bottom of this but had to give up due to other obligations. Although you were saying that _it was fine before_, it really wasnt. In any version of vue-meta there appears to be a memory leak if a global mixin adds a `head` function, which makes me believe the issue might originate from Vue.
The only 'issue' that seems to exists is that since vue-meta `v2` that memory leak is causing issues, might just be because something wasnt working before but since `v2` it is.

**Comment by t1gu1** [comment]
> **The only 'issue' that seems to exists is** that since vue-meta v2 that memory leak is causing issues, might just be **because something wasnt working before but since v2 it is**.

That's really make me laugh haha :rofl: 
---------------------------------------------------
@pimlie, I've two small question for you. (or everyone)

1. Is vue-meta a defaut Package installed with/inside Nuxt?
2. (If it's a default nuxt package) If I install the package, will it "ovewrite" the one used by Nuxt?

I just wanna to be sure, because I don't have that package in my `package.json`.

Thanks

**Comment by pimlie** [comment]
1. vue-meta is a dependency of nuxt indeed.
2. With `yarn upgrade` no. With `yarn add vue-meta@2.3.0` also no until nuxt upgrades to v3 but your package.json is still pointing to v2. But Nuxt will log a warning about version mismatching anyway, so yes but no.

See [here](https://docs.npmjs.com/misc/semver) for more info how package versions are matched

**Comment by t1gu1** [comment]
@danielroe (Just to share) I'm not sure about the point of strict on that issue. https://github.com/nuxt/nuxt.js/issues/6279#issuecomment-528763416

In the official documentation, they say:
> **Do not enable strict mode when deploying for production!**

Reference: https://vuex.vuejs.org/guide/strict.html

**Comment by AndrewBogdanovTSS** [comment]
@t1gu1 if you are using yarn you can add:
```
"resolutions": {
    "vue-meta": "2.3.0"
  }
```
inside your package.json - that will make yarn choose specified version over the once that is required by the related package

**Comment by manniL** [comment]
@t1gu1 it's not enabled in prod :wink:

**Comment by rafaelmagalhaes** [comment]
I also notice since the upgrade to 2.9.0 vue-meta was updated, and all of my configs inside  head: {} in nuxt.config.js render nothing, I fixed this by downgrading to version 2.8.1 which uses vue-meta 1.6 , I don't know if its an issue with nuxt or vue-meta, I also tried upgrading to nuxt latest and manually installing vue-meta version 1.6 but got an error saying version 2.0 is required 


here's my head config in nuxt.config.js
```
  head: {
    meta: [
      {name: 'viewport', content: 'width=device-width, initial-scale=1'},
      {name: 'referrer', content: 'unsafe-url'}
    ],
    link: [
      {rel: 'stylesheet', href: 'https://use.fontawesome.com/releases/v5.7.2/css/all.css'},
      {rel: 'stylesheet', href: 'https://fonts.googleapis.com/css?family=Open+Sans:400,700'}
    ]
  },
```

everything is missing after upgrading to version 2.9 or above

**Comment by pxwee5** [comment]
This looks like a bug to me, is there a way we can have those stuff disabled by default ?

**Comment by testn** [comment]
Has this been addressed?

**Comment by danielroe** [comment]
I'm closing this as there is no reproduction and we've now released Nuxt 3, which is significantly faster.
